google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableBatchingCallSettings.java

```diff
@@ -145,9 +188,92 @@ public Builder setRetrySettings(@Nonnull RetrySettings retrySettings) {
       return this;
     }
 
+    /**
+     * Enables / disables latency based throttling. If enabling the setting, targetRpcLatency needs
+     * to be set.
+     *
+     * <p>{@link BatchingSettings} must be set before calling this setting. Sets up a {@link
+     * FlowController} bases on isLatencyBasedThrottlingEnabled and BatchingSettings.
+     */
+    public Builder setLatencyBasedThrottling(
+        boolean isLatencyBasedThrottlingEnabled, @Nullable Long targetRpcLatency) {
+      Preconditions.checkState(
+          this.batchingSettings != null,
+          "batchingSettings must be set before enabling / disabling throttling");
+      Preconditions.checkArgument(
+          !isLatencyBasedThrottlingEnabled || targetRpcLatency != null,
+          "target RPC latency must be set if latency based throttling is enabled");
+      Preconditions.checkArgument(
+          targetRpcLatency == null || targetRpcLatency > 0,
+          "if target RPC latency is set, it must be greater than 0");
```

Comment:
Can this validation move to build() time? This way we wont force our users into set these in a particular order

Reply:
Removed the checks for BatchingSettings. These 2 are just checking if targetRpcLatency is valid